src/sage/matrix/matrix_cmr_sparse.pyx

```diff
@@ -128,6 +128,31 @@ cdef class Matrix_cmr_chr_sparse(Matrix_cmr_sparse):
         if immutable:
             self.set_immutable()
 
+    def matrix_from_rows_and_columns(self, rows, columns):
+        if not isinstance(rows, (list, tuple)):
+            rows = list(rows)
+
+        if not isinstance(columns, (list, tuple)):
+            columns = list(columns)
+
+        if cmr == NULL:
+            CMRcreateEnvironment(&cmr)
+
+        cdef CMR_SUBMAT *submatrix = NULL
+        cdef CMR_CHRMAT *cmr_submatrix = NULL
+
+        CMR_CALL(CMRsubmatCreate(cmr, len(rows), len(columns), &submatrix))
+
+        for i in range(submatrix.numRows):
+            submatrix.rows[i] = rows[i]
+
+        for j in range(submatrix.numColumns):
+            submatrix.columns[j] = columns[j]
+
+        CMR_CALL(CMRchrmatZoomSubmat(cmr, self._mat, submatrix, &cmr_submatrix))
+
+        return Matrix_cmr_chr_sparse._from_cmr(cmr_submatrix)
```

Comment:
`submatrix` needs to be freed, I think

Reply:
Does this mean to free the intermediate variable submatrix? I will fix the other issues and add a new commit